src/Interpreters/MonotonicityCheckVisitor.h

```diff
@@ -97,12 +102,30 @@ class MonotonicityCheckMatcher
         if (data.isRejected())
             return;
 
-        /// TODO: monotonicity for functions of several arguments
-        if (!ast_function.arguments || ast_function.arguments->children.size() != 1)
+        /// Monotonicity check only works for functions that contain at most two arguments and one of them must be a constant.
+        if (!ast_function.arguments)
+        {
+            data.reject();
+            return;
+        }
+
+        auto arguments_size =  ast_function.arguments->children.size();
+
+        if (arguments_size == 0 || arguments_size > 2)
         {
             data.reject();
             return;
         }
+        else if (arguments_size == 2)
+        {
+            /// If the function has two arguments, then one of them must be a constant.
+            if (!ast_function.arguments->children[0]->as<ASTLiteral>()
+                && !ast_function.arguments->children[1]->as<ASTLiteral>())
+            {
+                data.reject();
+                return;
+            }
```

Comment:
Could there be a specific error message for cases where the destination partition function is not supported?

In the following example `plus(B, A)` is monotonic, it doesn't work for the reason it's not supported. Hence, this can be confusing:
``` sql
CREATE TABLE source (A Int64, B Int64) engine=MergeTree ORDER BY A PARTITION BY plus(A, B);
CREATE TABLE destination (A Int64, B Int64) engine=MergeTree ORDER BY A PARTITION BY plus(B, A);
INSERT INTO TABLE source VALUES (1, 4), (4, 1);
ALTER TABLE destination ATTACH PARTITION ID '5' FROM source;
Received exception from server (version 23.11.1):
Code: 36. DB::Exception: Received from localhost:9000. DB::Exception: Destination table partition expression is not monotonically increasing. (BAD_ARGUMENTS)
(query: ALTER TABLE destination ATTACH PARTITION ID '5' FROM source;)
```

Maybe a specific boolean flag in `MonotonicityCheckMatcher::Data` and a matching exception message.

Reply:
I understand the motivation, but it does not look like it's in the scope of this PR. The existing MonotonicityChecker has a variety of `data.reject()` that might happen for unspecified reasons.

My suggestion is that we focus on the functionality itself and address it later (though it is arguable this is part of the functionality).

If you consider it a must, I can introduce an enum for each and every existing `data.reject()` so a corresponding exception is thrown on caller side.